nixos/modules/hardware/video/nvidia.nix

```diff
@@ -534,7 +534,12 @@ in {
 
         services.dbus.packages = lib.optional cfg.dynamicBoost.enable nvidia_x11.bin;
 
-        hardware.firmware = lib.optional cfg.open nvidia_x11.firmware;
+        hardware.firmware =
+          let
+            isOpen = cfg.open;
+            isNewUnfree = lib.versionAtLeast nvidia_x11.version "555";
+          in
+          lib.optional (isOpen || isNewUnfree) nvidia_x11.firmware;
```

Comment:
Using let in here, makes the code a little more complex than it needs to be. Also isOpen is not really any improvement over cfg.open